vars/parseStageInfo.groovy

```diff
@@ -252,19 +252,29 @@ def call(Map config = [:]) {
     if (startedByUser() && params.TestRepeat && params.TestRepeat != "") {
       repeat = params.TestRepeat
     } else {
-      // Next highest priority is a stage specific Test-repeat-*
-      repeat = cachedCommitPragma("Test-repeat" + result['pragma_suffix'], null)
-      if (!repeat) {
-        // Followed by the more general Test-repeat:
-        repeat = cachedCommitPragma("Test-repeat", null)
-      }
+      // Next highest priority is a stage specific Test-repeat-* then the general Test-repeat
+      repeat = cachedCommitPragma(
+        "Test-repeat" + result['pragma_suffix'], cachedCommitPragma("Test-repeat", null))
     }
     if (repeat) {
-      if (result['ftest_arg']) {
-        result['ftest_arg'] += " --repeat=" + repeat
-      } else {
-        result['ftest_arg'] += "--repeat=" + repeat
-      }
+      result['ftest_arg'] += " --repeat=" + repeat
+    }
+
+    String provider
+    // Highest priority is TestProvider parameter
+    if (startedByUser() && params.TestProvider && params.TestProvider != "") {
```

Comment:
Does using this parameter **really** want to be limited to jobs started (i.e. manually) by a user?  What if a job wants to run another job with this parameter set?

Reply:
This isn't any different than what we do with the `TestTag` and `TestRepeat` 'Build with Parameters' options.  It seems like that would be a larger PR.

Reply:
Since this PR is adding this specific code, I think it's fair to ask the question and possibly change the code depending on the answer and leaving the others to be updated organically if/when a consumer wants to change their behaviour.  This could perhaps happen sooner than later as we are starting to add the running of the daos pipeline as a testing stage of other pipelines.

Reply:
Done